### PR TITLE
Metrics Canister

### PR DESCRIPTION
**Pull request by TimWDev**

### CODE REVIEW

File: actors/metrics_service/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    pub payment_sagas_completed: u64,
+
+    pub send_sagas_created: u64,
+    pub send_sagas_failed: u64,
+    pub send_sagas_completed: u64,
+
+    pub outstanding_payment_sagas: BTreeMap<String, u64>,
+}
+
+pub enum PaymentSagaMetric {
```
Comment (SuddenlyHazel) [inline actors/metrics_service/src/metrics.rs:22]: Why do we need these? Can't we just react to the events directly?

File: actors/metrics_service/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    pub fn metric_account_created(&mut self) {
+        self.accounts_created += 1;
+    }
+
+    pub fn metric_money_moved(&mut self, amount: u64) {
+        self.total_money_moved += amount;
+    }
+
+    pub fn metric_payment_saga(&mut self, event: PaymentSagaMetric) {
```
Comment (SuddenlyHazel) [inline actors/metrics_service/src/metrics.rs:62]: RE above comment. Can't we just pass the event in here and match off that? Then we have all the data we need in a single place and a single data type.

File: actors/metrics_service/src/metrics.rs (hunk 1 above)
Comment (TimWDev) [reply actors/metrics_service/src/metrics.rs]: Fixed, was an event ownership issue

File: actors/metrics_service/src/metrics.rs (hunk 2 above)
Comment (TimWDev) [reply actors/metrics_service/src/metrics.rs]: Fixed, was an event ownership issue

**Review by SuddenlyHazel — APPROVED** [review]
Send it!

**Pull request merged**